src/rpc/ckb.rs

```diff
@@ -81,3 +83,148 @@ crate::jsonrpc!(pub struct CkbRpcClient {
+}
+
+impl CkbRpcClient {
+    pub fn get_packed_block(&mut self, hash: H256) -> Result<Option<JsonBytes>, crate::RpcError> {
+        self.post("get_block", (hash, Some(Uint32::from(0u32))))
+    }
+
+    // turn block response into BlcokView and cycle vec
+    fn blockresponse2view(
+        opt_resp: Option<BlockResponse>,
+    ) -> Result<(Option<BlockView>, Option<Vec<Cycle>>), crate::rpc::RpcError> {
+        if let Some(resp) = opt_resp {
+            match resp {
+                BlockResponse::Regular(block_view) => Ok((Some(block_view.get_value()?), None)),
+                BlockResponse::WithCycles(block_cycles) => {
+                    let block_view = block_cycles.block.get_value()?;
+                    let cycles = map(block_cycles.cycles);
+
+                    Ok((Some(block_view), cycles))
+                }
+            }
+        } else {
+            Ok((None, None))
+        }
+    }
+    /// Same as get_block except with parameter with_cycles and return BlockResponse
+    pub fn get_block_with_cycles(
+        &mut self,
+        hash: H256,
+    ) -> Result<(Option<BlockView>, Option<Vec<Cycle>>), crate::rpc::RpcError> {
+        let res = self.post::<_, Option<BlockResponse>>("get_block", (hash, None::<u32>, true))?;
+        Self::blockresponse2view(res)
+    }
+
+    // turn BlockResponse to JsonBytes and Cycle tupe
+    fn blockresponse2bytes(
+        opt_resp: Option<BlockResponse>,
+    ) -> Result<(Option<JsonBytes>, Option<Vec<Cycle>>), crate::rpc::RpcError> {
+        if let Some(resp) = opt_resp {
+            match resp {
+                BlockResponse::Regular(block_view) => {
+                    Ok((Some(block_view.get_json_bytes()?), None))
+                }
+                BlockResponse::WithCycles(block_cycles) => {
+                    let block_view = block_cycles.block.get_json_bytes()?;
+                    let cycles = map(block_cycles.cycles);
+                    Ok((Some(block_view), cycles))
+                }
+            }
+        } else {
+            Ok((None, None))
+        }
+    }
+
+    pub fn get_packed_block_with_cycles(
+        &mut self,
+        hash: H256,
+    ) -> Result<(Option<JsonBytes>, Option<Vec<Cycle>>), crate::rpc::RpcError> {
+        let res = self.post::<_, Option<BlockResponse>>(
+            "get_block",
+            (hash, Some(Uint32::from(0u32)), true),
+        )?;
+        Self::blockresponse2bytes(res)
+    }
+
+    /// Same as get_block_by_number except with parameter with_cycles and return BlockResponse
+    pub fn get_packed_block_by_number(
+        &mut self,
+        number: BlockNumber,
+    ) -> Result<Option<JsonBytes>, crate::rpc::RpcError> {
+        self.post("get_block_by_number", (number, Some(Uint32::from(0u32))))
+    }
+
+    pub fn get_block_by_number_with_cycles(
+        &mut self,
+        number: BlockNumber,
+    ) -> Result<(Option<BlockView>, Option<Vec<Cycle>>), crate::rpc::RpcError> {
```

Comment:
suggest to returning option tuple, this suggestion is applied to `get_packed_block_by_number_with_cycles` also:

```suggestion
    ) -> Result<Option<(BlockView, Vec<Cycle>)>, crate::rpc::RpcError> {
```